mempool/docs/config.md

```diff
@@ -0,0 +1,59 @@
+# Mempool Configuration
+
+Here we describe configuration options around mempool.
+For the purposes of this document, they are described
+as command-line flags, but they can also be passed in as
+environmental variables or in the config.toml file. The
+following are all equivalent:
+
+Flag: `--mempool.recheck_empty=false`
+
+Environment: `TM_MEMPOOL_RECHECK_EMPTY=false`
+
+Config:
+```
+[mempool]
+recheck_empty = false
+```
+
+
+## Recheck
+
+`--mempool.recheck=false` (default: true)
+
+`--mempool.recheck_empty=false` (default: true)
```

Comment:
@ebuchman Do we want to make this false by default?

mempool.go:368 suggests it is used for debugging

Reply:
Maybe. But currently the mempool is supposed to be configured so a correct proposer will only propose valid txs. Apps that use EndBlock could invalidate transactions in the mempool without the mempool knowing, and so a correct proposer could return invalid txs.

We just need to be clear either way. There's def a performance benefit to setting it false (ie. we won't replay all our mempool txs when an empty block is committed ...)